packages/connectivity/src/scp-cf/destination/destination.ts

```diff
@@ -324,9 +324,17 @@ const configMapping: Record<string, keyof Destination> = {
 };
 
 export function noDestinationErrorMessage(
-  destination: Destination | DestinationFetchOptions
+  destination: DestinationOrFetchOptionsXOR
 ): string {
   return isDestinationFetchOptions(destination)
     ? `Could not find a destination with name "${destination.destinationName}"! Unable to execute request.`
     : 'Could not find a destination to execute request against and no destination name has been provided (this should never happen)!';
 }
+
+type Without<T> = { [P in keyof T]?: never };
+type XOR<T, U> = (Without<T> & U) | (Without<U> & T);
```

Comment:
According to our [styleguide](https://github.com/SAP/cloud-sdk-js/blob/2.0/STYLEGUIDE.md#use-camel-case-for-acronyms) I think we should call this `Xor` - looks a bit weird, but then it would be consistent.

Reply:
BTW - if there are two keys that have the same name, this would still work, wouldn't it? (Unlikely, but I am asking out of interest).

Reply:
I tried this, and it doesn't work for keys with same name. Probably because in `Without<T>`, that property would be of type `never`. And it remains `never` in the resultant type `Without<T> & U` as well. 

It works when I use `Exclude` instead:
```
type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
type Xor<T, U> = (Without<T, U> & U) | (Without<U, T> & T);

export type DestinationOrFetchOptions = Xor<
  Destination,
  DestinationFetchOptions
>;
```

Should we consider handling this?

Reply:
I think this won't be necessary, but maybe add a comment, so that if this comes up in the future, we are aware?